base/broadcast.jl

```diff
@@ -549,17 +551,24 @@ combine_eltypes(f, A, As...) =
 """
     broadcast(f, As...)
 
-Broadcasts the arrays, tuples, `Ref`s and/or scalars `As` to a
-container of the appropriate type and dimensions. In this context, anything
-that is not a subtype of `AbstractArray`, `Ref` (except for `Ptr`s) or `Tuple`
-is considered a scalar. The resulting container is established by
-the following rules:
+Broadcast the function `f` over the arrays, tuples, collections, `Ref`s and/or scalars `As`.
+
+Broadcasting applies the function `f` over the elements of the container
+arguments and the scalars themselves in `As`. Singleton and missing dimensions
+are expanded to match the extents of the other arguments by virtually repeating
+the value. By default, only a limited number of types are considered scalars,
+including `Number`s, `Type`s and `Function`s; all other arguments are iterated
+over or indexed into elementwise.
+
+The resulting container type is established by the following rules:
 
- - If all the arguments are scalars, it returns a scalar.
```

Comment:
I imagine this rule still applies? Maybe worth keeping for clarity even if it's covered by more general rules below.